### PR TITLE
Ports over the CM13 Zippo lighter sounds

### PR DESCRIPTION
**Pull request by ChickyRooster**
## About The Pull Request
This adds in sounds for the zippo lighter for all those smokers on station! Sound ported from CM13

Commit adding this to CM13: https://github.com/cmss13-devs/cmss13/commit/549694f850dc4ebf05b478b7e1f2138b37910d8f

## How This Contributes To The Skyrat Roleplay Experience
More sounds for more immersion is always a good thing, and having sound play for the lighter adds to the overal atmosphere of the game.

## Proof of Testing

<details>
<summary>Screenshots/Videos

https://github.com/Skyrat-SS13/Skyrat-tg/assets/56152010/c05dc922-6d7e-4bcd-9d2e-1f0796b8cf1c

</summary>

</details>

## Changelog

:cl: AvianAviator, nauticall
Add: Added sounds to the Zippo lighter from CM13. Attributed to nauticall. 
/:cl:

### CODE REVIEW

File: code/game/objects/items/cigs_lighters.dm [hunk 1]
```diff
@@ -842,6 +842,7 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 		return ..()
 	if(lit)
 		set_lit(FALSE)
+		playsound(src, 'sound/items/zippo_close.ogg', 50, TRUE)
```
Comment (vinylspiders) [inline code/game/objects/items/cigs_lighters.dm]:
```suggestion
		playsound(src, 'sound/items/zippo_close.ogg', 50, TRUE) // SKYRAT EDIT ADDITION
```
These are nonmodular edits to core files so you need to mark them as such

File: code/game/objects/items/cigs_lighters.dm [hunk 2]
```diff
@@ -855,6 +856,7 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 		return
 
 	set_lit(TRUE)
+	playsound(src, 'sound/items/zippo_open.ogg', 50, TRUE)
```
Comment (vinylspiders) [inline code/game/objects/items/cigs_lighters.dm]:
```suggestion
	playsound(src, 'sound/items/zippo_open.ogg', 50, TRUE) // SKYRAT EDIT ADDITION
```

**Review by vinylspiders — CHANGES_REQUESTED** [review]
The 'zippo_close' sound is good but I'm not a big fan of the ending flame noises for the 'zippo_open' sound, do you think you could trim the audio to just the initial strike/click? 

It sounds almost like someone is blowing into a mic at the end to me which is the opposite of immersive.

Could do this in Audacity if you are unsure of where to start. It's pretty straightforward to do.

**Comment by ChickyRooster** [comment]
> It sounds almost like someone is blowing into a mic at the end to me which is the opposite of immersive.

I think you are mistaken by the cigarette being removed from the cigarette pack producing the bag ruffle sound, I 
should have made it more obvious or only foused on the lighter instead, 

Here's a video showcasing just the lighter, is this the sound you wanted changed?

https://github.com/Skyrat-SS13/Skyrat-tg/assets/56152010/86e50911-dad7-4ac4-882b-e13e000dc63a

**Comment by vinylspiders** [comment]
> > It sounds almost like someone is blowing into a mic at the end to me which is the opposite of immersive.
> 
> I think you are mistaken by the cigarette being removed from the cigarette pack producing the bag ruffle sound, I should have made it more obvious or only foused on the lighter instead,
> 
> Here's a video showcasing just the lighter, is this the sound you wanted changed?
> Communist.Garden.Bravo.2023-11-05.15-17-18.mp4

You are absolutely correct. I just checked the sound files and they are in fact, fine. The first video threw me off with whatever that noise was.  No need to change anything, disregard the comment about that.

That said, one last thing I am noticing: you will want to put the sound files in `modular_skyrat/master_files/sound/items` instead of the TG folder they are in currently (and change the filepath in the code accordingly).

**Comment by ChickyRooster** [comment]
Tested and changed, file paths should be all in order now :)

**Review by vinylspiders — APPROVED** [review]

**Comment by YakumoChen** [comment]
Please give proper attribution by linking the PR adding this to cm13

Also consider if this would benefit upstream (and us and all other downstreams by extension)

**Comment by ChickyRooster** [comment]
> Please give proper attribution by linking the PR adding this to cm13
> 
> Also consider if this would benefit upstream (and us and all other downstreams by extension)

While I could not find a PR, original commit has been linked, considered it as well.

**Comment by vinylspiders** [comment]
> > Please give proper attribution by linking the PR adding this to cm13
> > Also consider if this would benefit upstream (and us and all other downstreams by extension)
> 
> While I could not find a PR, original commit has been linked, considered it as well.

All good. In this case since it's not really possible to find that and because it's a port it is fine.

**Comment by desertmote** [comment]
This sfx was added to all lighters instead of just the zippo.

**Comment by ghost** [comment]
> This sfx was added to all lighters instead of just the zippo.

love to see it.

**Comment by vinylspiders** [comment]
> > This sfx was added to all lighters instead of just the zippo.
> 
> love to see it.

That is because for some reason, the zippo lighter is the basetype and all the other kinds of lighters are subtypes of it. Not how you'd expect it to be. https://github.com/Skyrat-SS13/Skyrat-tg/pull/24854

It is not a big deal in the grand scheme of things. With only two active maintainers for the whole repo we are bound to sometimes miss little things like this every now and then.

**Pull request merged**